### ISSUE TITLE
Incorrect uuid marshalling with embeded struct

### ISSUE THREAD

**Issue by hi019** [goccy/go-json#318]
```go
package main

import (
	"fmt"

	"github.com/goccy/go-json"
	"github.com/google/uuid"
)

type config struct {
	debug bool
	log func(...interface{})
}

type res struct {
	config `json:"-"`
	ID uuid.UUID `json:"id,omitempty"`
}

func main() {
	u := uuid.New()

	test := res{
		ID: u,
	}
    // uuid should be the same in both print statements
	fmt.Println(test.ID)

	b, _ := json.Marshal(test)
	fmt.Println(string(b))
}
```

Output:

```
a6209e56-60b1-4c3a-8fc0-fb2a3ea5d8b5
{"id":"00000000-0000-0000-0000-000000000000"}
```

**Comment by goccy** [comment]
Thank you for reporting ! I will fix this problem with #319

**Comment by hi019** [comment]
@goccy Great! Thanks for the fast fix :)

**Comment by goccy** [comment]
I released this fixing as [v0.9.1](https://github.com/goccy/go-json/releases/tag/v0.9.1) :)
